NitroxModel/Packets/Exceptions/UncorrelatedPacketException.cs

```diff
@@ -0,0 +1,36 @@
+﻿using System;
+using System.Runtime.Serialization;
+using JetBrains.Annotations;
+
+namespace NitroxModel.Packets.Exceptions
+{
+    public class UncorrelatedPacketException : Exception
+    {
+        public CorrelatedPacket InvalidPacket { get; }
+        public string ExpectedCorrelationId { get; }
+
+        public UncorrelatedPacketException(CorrelatedPacket invalidPacket, string expectedCorrelationId)
+        {
+            InvalidPacket = invalidPacket;
+            ExpectedCorrelationId = expectedCorrelationId;
+        }
+
+        public UncorrelatedPacketException(string message, CorrelatedPacket invalidPacket, string expectedCorrelationId) : base(message)
+        {
+            InvalidPacket = invalidPacket;
+            ExpectedCorrelationId = expectedCorrelationId;
+        }
+
+        public UncorrelatedPacketException(string message, Exception innerException, CorrelatedPacket invalidPacket, string expectedCorrelationId) : base(message, innerException)
+        {
+            InvalidPacket = invalidPacket;
+            ExpectedCorrelationId = expectedCorrelationId;
+        }
+
+        protected UncorrelatedPacketException([NotNull] SerializationInfo info, StreamingContext context, CorrelatedPacket invalidPacket, string expectedCorrelationId) : base(info, context)
```

Comment:
We are not using annotations elsewhere, do we want to use them?

Interesting, these are bundled with the game.

Reply:
That was auto-generated by the IDE.